icu4j/main/tests/core/src/com/ibm/icu/dev/test/impl/UnitsTest.java

```diff
@@ -686,7 +686,8 @@ public String toString() {
         }
 
         for (TestCase testCase : tests) {
-            UnitsRouter router = new UnitsRouter(testCase.inputUnit.second, testCase.region, testCase.usage);
+            UnitsRouter router = new UnitsRouter(testCase.inputUnit.second, ULocale.forLanguageTag(testCase.region),
```

Comment:
Same problem with forLanguageTag from just a reason. Look for all those cases.

Reply:
This is an internal class constructor that is only used for exposing UnitsRouter to the end users. Therefore, it is okay to leave this constructor taking the locale as a string. I used new Ulocale(locale).

The other constructor, which used for NumberFormatter, take the locale as it is provided by the users.

Reply:
That's not what Mark is talking about (and with "reason" he meant "region"). See above, and see my new comments.

Reply:
This code is slightly different now but still doesn't work. I am resolving this comment in favor of newer ones.